packages/core/src/render3/node_selector_matcher.ts

```diff
@@ -34,21 +34,33 @@ function isCssClassMatching(
   ngDevMode &&
       assertEqual(
           cssClassToMatch, cssClassToMatch.toLowerCase(), 'Class name expected to be lowercase.');
-  let i = 0;
+      let i = 0;
+  // Indicates whether we are processing value from the implicit
+  // attribute section (i.e. before the first marker in the array).
+  let isImplicitAttrsSection = true;
   while (i < attrs.length) {
-    let item = attrs[i++];
-    if (isProjectionMode && item === 'class') {
-      item = attrs[i] as string;
-      if (classIndexOf(item.toLowerCase(), cssClassToMatch, 0) !== -1) {
-        return true;
+    const item = attrs[i++];
```

Comment:
Let's minimize the diff (see also a comment below):
```suggestion
    let item = attrs[i++];
```